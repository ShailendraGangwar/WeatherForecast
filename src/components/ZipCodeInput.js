import React, { Component } from "react";
import { View, Button, StyleSheet, TextInput, TouchableOpacity, Text } from "react-native";

class ZipCodeInput extends Component {
  state = {
    zipCode: ""
  };


  zipCodeChangedHandler = val => {
    this.setState({
      zipCode: val
    });
  };

  zipCodeSubmitHandler = () => {
    this.props.onZipCodeSubmit(this.state.zipCode);
  };

  render() {
    return (
      <View style={styles.inputContainer}>
        <TextInput
          placeholder="Enter Zip Code (India)"
          value={this.state.zipCode}
          onChangeText={this.zipCodeChangedHandler}
          style={styles.zipInput}
          maxLength={6}
        />
        <TouchableOpacity
          style={styles.searchButton}
          onPress={this.zipCodeSubmitHandler}
        >
          <Text> Search </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  inputContainer: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  zipInput: {
    width: "60%",
    margin: 10,
    borderColor: 'red',
    borderWidth: 1,
    height: 30,
    borderRadius: 10,
    padding: 5
  },
  searchButton: {
    width: "30%",
    borderRadius: 10,
    borderColor: 'red',
    borderWidth: 1,
    height: 30,
    alignItems: "center",
    justifyContent: "center",
    margin: 10,
    backgroundColor: "#63c0e8",

  }
});

export default ZipCodeInput;
