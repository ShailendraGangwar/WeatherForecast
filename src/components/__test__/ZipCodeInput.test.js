import 'react-native';
import React from 'react'
import ZipCodeInput from '../ZipCodeInput';

import renderer from 'react-test-renderer';

it('should renders', () => {
  const tree = renderer.create(<ZipCodeInput />).toJSON();
  expect(tree).toMatchSnapshot();
});