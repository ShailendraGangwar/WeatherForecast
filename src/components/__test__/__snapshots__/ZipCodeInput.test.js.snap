// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={6}
    onChangeText={[Function]}
    placeholder="Enter Zip Code (India)"
    style={
      Object {
        "borderColor": "red",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 30,
        "margin": 10,
        "padding": 5,
        "width": "60%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "red",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 30,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "width": "30%",
      }
    }
  >
    <Text>
       Search 
    </Text>
  </View>
</View>
`;
